### ISSUE TITLE
Publish action image to public registry

### ISSUE THREAD

**Issue by 20k-ultra** [balena-io/deploy-to-balena-action#27]
If we publish a built image of the action then we can reference that in workflows which would save the entire build time. This will speed up the action significantly.

https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#example-using-a-docker-hub-action

**Comment by klutchell** [comment]
We should attempt to use the balena registry and an open fleet for this once we add support for vanity image names.
More to come.

**Comment by 20k-ultra** [comment]
Can you elaborate on the open fleet part ? I like the idea of using balena registry as long as the image path is recognizable and not random strings. People have to add this url to their workflow so if it's random strings then the UX seems worse compared to a recognizable location.

**Comment by klutchell** [comment]
https://jel.ly.fish/milestone-block-images-balena-registry-4ef9934

It's a blocks milestone for now, but it impacts any project we would normally put in DH.

> We want to host balenaBlock images in the balena registry, but have human-readable references for users to add to their docker-compose file.

**tl;dr** we are looking at adding a pre-pull hook to our registry to translate names to full image paths.

**Comment by klutchell** [comment]
Now that the proxy has shipped we should push this project to a fleet (via deploy-to-balena action of course) and have users pull the image at runtime to save compiling.

```
      - uses: bh.cr/balena/deploy-to-balena@v0.9.0
        id: build
        with:
          balena_token: ${{ secrets.BALENA_TOKEN }}
          fleet: my_org/sample_fleet
```

**Comment by 20k-ultra** [comment]
I'd rather push to Github's registry tbh because I don't want to maintain the fleet on balenacloud

**Comment by ab77** [comment]
> > because I don't want to maintain the fleet on balenacloud
> 
> this is concerning on many levels

Indeed. To resolve this issue (correctly), we first need to publish this code as a block on balenaHub. Once it's there, we can reference it in GitHub actions worflows using normal docker run syntax, straight from bh.cr/.

**Comment by 20k-ultra** [comment]
It seems like really unnecessary dogfooding. Just build the image and push it to ghcr.

The outcome seems the same with extra steps if we push to bh.cr.

**Comment by 20k-ultra** [comment]
The action is pushing to gchr since https://github.com/balena-io/deploy-to-balena-action/commit/411241f6afa6c87f413acc5bbc9660c706a18ed5

**Comment by klutchell** [comment]
> The action is pushing to gchr since [411241f](https://github.com/balena-io/deploy-to-balena-action/commit/411241f6afa6c87f413acc5bbc9660c706a18ed5)

How do we use it though? I tried calling the docker image from my workflow and it can't find the inputs.

**Comment by klutchell** [comment]
This PR might help https://github.com/balena-io/deploy-to-balena-action/pull/195

**Comment by 20k-ultra** [comment]
> > The action is pushing to gchr since [411241f](https://github.com/balena-io/deploy-to-balena-action/commit/411241f6afa6c87f413acc5bbc9660c706a18ed5)
> 
> How do we use it though? I tried calling the docker image from my workflow and it can't find the inputs.

check out https://github.com/20k-ultra/action-tester/actions/runs/2116842116/workflow

Doing so however revealed https://github.com/balena-io/deploy-to-balena-action/issues/147
